force-app/main/default/classes/FieldSetService_TEST.cls

```diff
@@ -7,54 +7,138 @@
+            );
+
+            System.assertEquals(
+                fieldDescribe.isAccessible(),
+                fieldsResultByApiName.get(fieldSetMember.getFieldPath())
+                    .get('isAccessible')
+            );
         }
 
         System.assertEquals(fieldSet.getFields().size(), fieldsResult.size());
     }
+
+    @IsTest
+    private static void shouldSkipUnsupportedFieldTypes() {
+        Test.startTest();
+
+        FieldSetService fieldSetService = new FieldSetService();
+        fieldSetService.unsupportedFieldTypes = new Set<String>{ 'DATE' };
+        Schema.FieldSet fieldSet = Schema.SObjectType.ProgramEngagement__c.FieldSets.CreateProgramEngagement;
+
+        List<Map<String, Object>> fieldsResult = fieldSetService.getFieldSetForLWC(
+            String.valueOf(ProgramEngagement__c.SObjectType),
+            fieldSet.getName()
+        );
+
+        Test.stopTest();
+
+        Map<String, Map<String, Object>> fieldsResultByApiName = new Map<String, Map<String, Object>>();
+        for (Map<String, Object> fieldMap : fieldsResult) {
+            fieldsResultByApiName.put(String.valueof(fieldMap.get('apiName')), fieldMap);
+        }
+
+        for (Schema.FieldSetMember fieldSetMember : fieldSet.getFields()) {
+            if (fieldSetMember.getType().name() == 'DATE') {
+                System.assert(
+                    !fieldsResultByApiName.containsKey(fieldSetMember.getFieldPath()),
+                    'The field should have been skipped: ' + fieldSetMember.getFieldPath()
+                );
+                continue;
+            }
+
+            System.assert(
+                fieldsResultByApiName.containsKey(fieldSetMember.getFieldPath()),
+                'The field should not have been skipped: ' + fieldSetMember.getFieldPath()
+            );
+        }
+
+        System.assertNotEquals(
+            fieldSet.getFields().size(),
+            fieldsResult.size(),
+            'Expected the fieldsResult size to be less than the fieldSet size.'
+        );
+    }
+
+    @IsTest
+    private static void shouldReturnCompoundFieldDescribe() {
+        String expectedFieldName = String.valueOf(Contact.Name);
+        DescribeFieldResult actualField = new FieldSetService()
+            .getCompoundFieldDescribe(
+                SObjectType.Contact,
+                SObjectType.Contact.fields.FirstName
+            );
+
+        System.assertEquals(
+            expectedFieldName,
+            actualField.getName(),
+            'Expected the Compound Field to be returned.'
+        );
+    }
+
+    @IsTest
+    private static void shouldReturnUILabel() {
+        Map<DescribeFieldResult, String> expectedLabelByField = new Map<DescribeFieldResult, String>{
+            SObjectType.Contact.fields.AccountId => 'Account Name',
+            SObjectType.Opportunity.fields.ContactId => 'Full Name',
+            SObjectType.Account.fields.OwnerId => 'Owner'
```

Comment:
should we use the same references for the values here or does that end up being an A=A test?

Reply:
Totally, I can't think of any reason not to.